pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
-		output = os.Stdout
+// fetchAndReset is a convenience method that fetches updates from the remote repository
+// for a specific branch, and then resets the current branch to a specified commit.
+func (r *repository) fetchAndReset(branch string) error {
+	if err := r.fetch(branch); err != nil {
+		return fmt.Errorf("fetchAndReset failure: %w", err)
 	}
-	return g.gitCmd(output, args...)
-}
 
-func (g *git) gitOutput(args ...string) (string, error) {
-	output := &bytes.Buffer{}
-	err := g.gitCmd(output, args...)
-	return strings.TrimSpace(output.String()), err
+	return r.hardReset(branch)
 }
 
-func (g *git) setCredential(cred *corev1.Secret) error {
-	if cred == nil {
-		return nil
+// updateRefSpec updates the reference specification (RefSpec) in the fetch options
+// of the repository operation.
+//   - If a branch name is provided, it sets the RefSpec to fetch that specific branch.
+//   - Otherwise, it sets the RefSpec to fetch all branches.
+//
+// fetching the last commit of one branch is faster than fetching from all branches.
+func (r *repository) updateRefSpec(branch string) {
+	var newRefSpec string
+
+	if branch != "" {
+		newRefSpec = fmt.Sprintf("+refs/heads/%s:refs/remotes/origin/%s", branch, branch)
+	} else {
+		newRefSpec = "+refs/heads/*:refs/remotes/origin/*"
 	}
 
-	if cred.Type == corev1.SecretTypeBasicAuth {
-		username, password := cred.Data[corev1.BasicAuthUsernameKey], cred.Data[corev1.BasicAuthPasswordKey]
-		if len(password) == 0 && len(username) == 0 {
-			return nil
-		}
+	if len(r.fetchOpts.RefSpecs) > 0 {
+		r.fetchOpts.RefSpecs[0] = config.RefSpec(newRefSpec)
+	} else {
+		r.fetchOpts.RefSpecs = []config.RefSpec{config.RefSpec(newRefSpec)}
+	}
+}
 
-		u, err := url.Parse(g.URL)
-		if err != nil {
-			return err
-		}
-		u.User = url.User(string(username))
-		g.URL = u.String()
-		g.password = string(password)
-	} else if cred.Type == corev1.SecretTypeSSHAuth {
-		key, err := ssh.ParseRawPrivateKey(cred.Data[corev1.SSHAuthPrivateKey])
-		if err != nil {
-			return err
-		}
-		sshAgent := agent.NewKeyring()
-		err = sshAgent.Add(agent.AddedKey{
-			PrivateKey: key,
-		})
-		if err != nil {
-			return err
-		}
-		g.knownHosts = cred.Data["known_hosts"]
-		g.agent = &sshAgent
+// fetch fetches updates from the remote repository for a specific branch.
+// If the fetch operation is already up-to-date, this is not treated as an error.
+// Any other error that occurs during fetch is returned.
+func (r *repository) fetch(branch string) error {
+	r.updateRefSpec(branch)
+	fetchOptions := r.fetchOpts
+
+	err := r.repoGogit.Fetch(fetchOptions)
+	if err != nil && err != gogit.NoErrAlreadyUpToDate {
+		return fmt.Errorf("fetch failure: %w", err)
 	}
 
 	return nil
 }
 
-func (g *git) clone(branch string) error {
-	gitDir := filepath.Join(g.Directory, ".git")
-	if dir, err := os.Stat(gitDir); err == nil && dir.IsDir() {
-		return nil
-	}
+func (r *repository) checkout(branch plumbing.ReferenceName) (plumbing.Hash, error) {
```

Comment:
Is this function used anywhere ? If not can we remove this ? It will be dead code.

Reply:
I have implemented a fix, please resolve the conversation.

I have removed, however this function will be used in the next PR (just stating, it should not be on this PR, you are right about it)